axelor-account/src/main/java/com/axelor/apps/account/service/payment/paymentsession/PaymentSessionValidateServiceImpl.java

```diff
@@ -1111,4 +1143,259 @@ protected void recomputeAmountPaid(MoveLine moveLine) {
+              null,
+              this.getAccountingDate(paymentSession, invoiceTerm),
+              invoiceTerm.getBankDetails());
+
+      paymentAmountMap.put(move, BigDecimal.ZERO);
+    }
+
+    boolean out = paymentSession.getPaymentMode().getInOutSelect() == PaymentModeRepository.OUT;
+
+    MoveLine moveLine =
+        this.generateMoveLine(
+            move,
+            invoiceTerm.getPlacementMoveLine().getPartner(),
+            invoiceTerm.getPlacementMoveLine().getAccount(),
+            invoiceTerm.getAmountPaid(),
+            invoiceTerm.getPlacementMoveLine().getOrigin(),
+            this.getMoveLineDescription(paymentSession),
+            out);
+
+    Account account =
+        paymentModeService.getPaymentModeAccount(
+            paymentSession.getPaymentMode(),
+            paymentSession.getCompany(),
+            paymentSession.getBankDetails(),
+            false);
+
+    MoveLine cashPartMoveLine =
+        move.getMoveLineList().stream()
+            .filter(ml -> ml.getAccount().equals(account))
+            .findFirst()
+            .orElse(null);
+
+    if (cashPartMoveLine != null) {
+
+      cashPartMoveLine = updateMoveLineAmounts(cashPartMoveLine, invoiceTerm);
+
+    } else {
+      cashPartMoveLine =
+          this.generateMoveLine(
+              move,
+              null,
+              account,
+              invoiceTerm.getAmountPaid(),
+              invoiceTerm.getMoveLine().getOrigin(),
+              this.getMoveLineDescription(paymentSession),
+              !out);
+    }
+
+    if (paymentAmountMap.get(move) != null) {
+      paymentAmountMap.replace(move, cashPartMoveLine.getCurrencyAmount());
+    }
+
+    move.setDescription(
+        this.getMoveDescription(paymentSession, cashPartMoveLine.getCurrencyAmount())
+            .concat(String.format(" - %s", I18n.get("Payment"))));
+
+    moveComputeService.autoApplyCutOffDates(move);
+
+    Reconcile reconcile =
+        reconcileService.reconcile(
+            moveLine, invoiceTerm.getPlacementMoveLine(), null, false, false);
+
+    invoiceTermService.payInvoiceTerms(invoiceTerm.getPlacementMoveLine().getInvoiceTermList());
+    invoiceTermService.payInvoiceTerms(moveLine.getInvoiceTermList());
+
+    reconcileService.updatePayment(
+        reconcile,
+        invoiceTerm.getMoveLine(),
+        invoiceTerm.getInvoice(),
+        invoiceTerm.getMoveLine().getMove(),
+        move,
+        invoiceTerm.getAmountPaid(),
+        false);
+  }
+
+  protected MoveLine updateMoveLineAmounts(MoveLine moveLine, InvoiceTerm invoiceTerm) {
+    if (moveLine == null || invoiceTerm == null) {
+      return null;
+    }
+
+    if (moveLine.getCredit().signum() != 0) {
+      moveLine.setCredit(moveLine.getCredit().add(invoiceTerm.getAmountPaid()));
+
+    } else {
+      moveLine.setDebit(moveLine.getDebit().add(invoiceTerm.getAmountPaid()));
+    }
+    moveLine.setCurrencyAmount(moveLine.getCurrencyAmount().add(invoiceTerm.getAmountPaid()));
```

Comment:
I don't know if this is supposed to work with multi currency but if it does there might be an issue here